backend/tests/Designer.Tests/GiteaIntegrationTests/GiteaIntegrationTestsBase.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using Designer.Tests.Controllers.ApiTests;
+using Designer.Tests.Fixtures;
+using DotNet.Testcontainers.Builders;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.Mvc.Testing.Handlers;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace Designer.Tests.GiteaIntegrationTests
+{
+    [Collection(nameof(GiteaCollection))]
+    public abstract class GiteaIntegrationTestsBase<TController, TControllerTest> : ApiTestsBase<TController, TControllerTest>, IDisposable
+        where TController : ControllerBase
+        where TControllerTest : class
+    {
+        protected readonly GiteaFixture GiteaFixture;
+
+        protected string CreatedFolderPath { get; set; }
+
+        private CookieContainer CookieContainer { get; } = new CookieContainer();
+
+        /// <summary>
+        /// Used when performing chained calls to designer api
+        /// </summary>
+        protected void InvalidateAllCookies()
+        {
+            foreach (Cookie cookie in CookieContainer.GetAllCookies())
+            {
+                cookie.Expires = DateTime.Now - TimeSpan.FromHours(1);
+            }
+        }
+
+        public void Dispose()
+        {
+            if (!string.IsNullOrWhiteSpace(CreatedFolderPath) && Directory.Exists(CreatedFolderPath))
+            {
+                var directory = new DirectoryInfo(CreatedFolderPath) { Attributes = FileAttributes.Normal };
+
+                foreach (var info in directory.GetFileSystemInfos("*", SearchOption.AllDirectories))
+                {
+                    info.Attributes = FileAttributes.Normal;
+                }
+
+                directory.Delete(true);
+            }
+        }
+
+        protected override void ConfigureTestServices(IServiceCollection services)
+        {
+
+        }
+
+        protected GiteaIntegrationTestsBase(WebApplicationFactory<TController> factory, GiteaFixture giteaFixture) : base(factory)
+        {
+            GiteaFixture = giteaFixture;
+        }
+
+        protected override HttpClient GetTestClient()
+        {
+            string configPath = GetConfigPath();
+
+            var client = Factory.WithWebHostBuilder(builder =>
+            {
+                builder.ConfigureAppConfiguration((_, conf) =>
+                {
+                    conf.AddJsonFile(configPath);
+                    conf.AddJsonStream(GenerateGiteaOverrideConfigStream());
+                });
+
+                builder.ConfigureTestServices(ConfigureTestServices);
+            }).CreateDefaultClient(new GiteaAuthDelegatingHandler(GiteaFixture.GiteaUrl), new CookieContainerHandler(CookieContainer));
```

Comment:
Can this be fixed by applying `using` before usage?

Reply:
Should not be disposed til the end of test execution. Can be extracted as a field and added in dispose method of the class but it's not really relevant since those are tests classes.